compiler/rustc_parse/src/parser/item.rs

```diff
@@ -81,6 +81,11 @@ impl<'a> Parser<'a> {
                         span,
                         "consider using `const` or `static` instead of `let` for global variables",
                     );
+                } else if self.parse_const_block(span, false).is_ok() {
```

Comment:
This logic should probably be moved to `parse_item` (well, one of its helpers like `parse_item_common`) since this is really an item recovery.

Reply:
should I just yeet all these suggestions that are secretly item recoveries over there?

Reply:
If you'd like to do so, please do.

Reply:
yeah, I think it's important that both of these suggestions continue to live next to each other so they get refactored together.